allauth/socialaccount/providers/oauth2/tests.py

```diff
@@ -145,64 +112,80 @@ def setUp(self):
-        self.assertEqual(
-            proxy_response['location'],
-            ('https://tweedledee/fake/login/callback/'
-            '?process=login&state=%s' % urlquote(serialized_state)))
+        expected_url = 'https://tweedledee/fake/login/callback/'
+        actual_url = proxy_response['location']
+        self.assertTrue(actual_url.startswith(expected_url))
+        self.assertEqual(self.param('process', actual_url), 'login')
+        self.assertEqual(json.loads(self.param('state', actual_url)), state)
 
     def tests_redirects_request_with_domain_whitelisted_host(self):
         state = {'host': 'https://foo.sub.domain.com'}
         serialized_state = json.dumps(state)
-        self.init_request(
-            'fake_proxy', dict(process='login', state=serialized_state))
+        request = self.init_request(
+            'fake_proxy',
+            dict(process='login', state=serialized_state),
+        )
         proxy_response = proxy_login_callback(
-            self.request, callback_view_name='fake_callback')
+            request,
+            callback_view_name='fake_callback',
+        )
         self.assertEqual(proxy_response.status_code, 302)  # Redirect
-        self.assertEqual(
-            proxy_response['location'],
-            ('https://foo.sub.domain.com/fake/login/callback/'
-            '?process=login&state=%s' % urlquote(serialized_state)))
+        expected_url = 'https://foo.sub.domain.com/fake/login/callback/'
+        actual_url = proxy_response['location']
+        self.assertTrue(actual_url.startswith(expected_url))
+        self.assertEqual(self.param('process', actual_url), 'login')
+        self.assertEqual(json.loads(self.param('state', actual_url)), state)
 
     def test_rejects_request_with_scheme_mismatch(self):
         state = {'host': 'ftp://tweedledee'}
-        self.init_request(
-            'fake_proxy', dict(process='login', state=json.dumps(state)))
+        request = self.init_request(
+            'fake_proxy',
+            dict(process='login', state=json.dumps(state)),
+        )
         with self.assertRaises(PermissionDenied):
             proxy_login_callback(
-                self.request, callback_view_name='fake_callback')
+                request,
+                callback_view_name='fake_callback',
+            )
 
     def test_rejects_request_with_whitelisted_prefix(self):
         state = {'host': 'https://tweedledee.creds4u.biz'}
-        self.init_request(
-            'fake_proxy', dict(process='login', state=json.dumps(state)))
+        request = self.init_request(
+            'fake_proxy',
+            dict(process='login', state=json.dumps(state))
+        )
         with self.assertRaises(PermissionDenied):
             proxy_login_callback(
-                self.request, callback_view_name='fake_callback')
+                request,
+                callback_view_name='fake_callback',
+            )
```

Comment:
The other indentation changes we triggered by linting with the `checkqa` tox command. This brings these changes up to date with the expectations for the upstream repo